inc/class-cas.php

```diff
@@ -233,15 +239,34 @@ public function authenticate( $user, $username, $password ) {
 					}
 				}
 			}
+			$message = $this->authenticationFailedMessage( $this->options['provision'] );
 			if ( $this->forcedRedirection ) {
-				wp_die( __( 'CAS authentication failed.', 'pressbooks-cas-sso' ) );
+				wp_die( $message );
 			} else {
-				return new \WP_Error( 'authentication_failed', __( 'CAS authentication failed.', 'pressbooks-cas-sso' ) );
+				return new \WP_Error( 'authentication_failed', $message );
 			}
 		}
 		return null;
 	}
 
+	/**
+	 * @param string $provision
+	 *
+	 * @see \Aldine\Helpers\handle_contact_form_submission for $email code
+	 *
+	 * @return string
+	 */
+	public function authenticationFailedMessage( $provision ) {
+		if ( $provision === 'refuse' ) {
+			$email = get_blog_option( get_main_site_id(), 'pb_network_contact_email', get_site_option( 'admin_email', '' ) );
+			$email = ( ! empty( $email ) ? ": {$email}" : '.' );
+			$message = sprintf( __( "Unable to log in: You do not have an account on this Pressbooks network. To request an account, please contact your institution's Pressbooks Network Manager", 'pressbooks-cas-sso' ), $email );
```

Comment:
Where is the `$email` value used in this function?

Reply:
Doh. Thanks.